cmd/fs-v1-multipart.go

```diff
@@ -392,32 +389,33 @@ func (fs *FSObjects) ListObjectParts(bucket, object, uploadID string, partNumber
 		if entry == fsMetaJSONFile {
 			continue
 		}
-		partNumber, etag1, err := fs.decodePartFile(entry)
-		if err != nil {
-			return result, toObjectErr(errors.Trace(err))
+		partNumber, etag1, derr := fs.decodePartFile(entry)
+		if derr != nil {
+			return result, toObjectErr(errors.Trace(derr))
 		}
 		etag2, ok := partsMap[partNumber]
 		if !ok {
 			partsMap[partNumber] = etag1
 			continue
 		}
-		stat1, err := fsStatFile(pathJoin(uploadIDDir, fs.encodePartFile(partNumber, etag1)))
-		if err != nil {
-			return result, toObjectErr(errors.Trace(err))
+		stat1, serr := fsStatFile(pathJoin(uploadIDDir, fs.encodePartFile(partNumber, etag1)))
+		if serr != nil {
+			return result, toObjectErr(errors.Trace(serr))
 		}
-		stat2, err := fsStatFile(pathJoin(uploadIDDir, fs.encodePartFile(partNumber, etag2)))
-		if err != nil {
-			return result, toObjectErr(errors.Trace(err))
+		stat2, serr := fsStatFile(pathJoin(uploadIDDir, fs.encodePartFile(partNumber, etag2)))
+		if serr != nil {
+			return result, toObjectErr(errors.Trace(serr))
 		}
 		if stat1.ModTime().After(stat2.ModTime()) {
 			partsMap[partNumber] = etag1
 		}
 	}
+
 	var parts []PartInfo
 	for partNumber, etag := range partsMap {
 		parts = append(parts, PartInfo{PartNumber: partNumber, ETag: etag})
 	}
-	sort.SliceStable(parts, func(i int, j int) bool {
+	sort.Slice(parts, func(i int, j int) bool {
```

Comment:
why change to `Slice`?

Reply:
Because existing code is actually wrong. We shouldn't be using Stable sorts unless we have more than 1 slice to sort from.

Reply:
okay

Reply:
Will send a separate PR for this.

Reply:
`SliceStable` is about keeping equal elements in the same order as before. Since the elements are expected to be distinct here, this does not matter.

Reply:
https://github.com/minio/minio/pull/5586 sent a PR for this..